From 3bfc2e75b3f399b6bf0e903b11e139ca6bfb77f3 Mon Sep 17 00:00:00 2001
From: Mingli Yu <mingli.yu@windriver.com>
Date: Wed, 26 Oct 2022 12:29:18 +0800
Subject: [PATCH] nscd.h: explicit use the internal __time64_t type

we do build nss modules with 64 time_t, however some features are built
on libc.so itself and in such cases we need to explicit use the internal
__time64_t type.

Upstream-Status: Backport [https://sourceware.org/pipermail/libc-alpha/2022-October/142938.html]

Signed-off-by: Mingli Yu <mingli.yu@windriver.com>
---
 nscd/nscd.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nscd/nscd.h b/nscd/nscd.h
index 368091aef8..f15321585b 100644
--- a/nscd/nscd.h
+++ b/nscd/nscd.h
@@ -65,7 +65,7 @@ typedef enum
 struct traced_file
 {
   /* Tracks the last modified time of the traced file.  */
-  time_t mtime;
+  __time64_t mtime;
   /* Support multiple registered files per database.  */
   struct traced_file *next;
   int call_res_init;
-- 
2.25.1

